### PR TITLE
[CXF-7910] Change JAX-WS javax to jakarta artifact dependencies

### PR DESCRIPTION
**Pull request by deki**

### CODE REVIEW

**Comment by deki** [comment]
It's just the flaky test failing, so I think this is ready for merge...

File: parent/pom.xml [hunk 1]
```diff
@@ -2854,35 +2878,38 @@
...
+                <dependency>
+                    <groupId>jakarta.xml.soap</groupId>
+                    <artifactId>jakarta.xml.soap-api</artifactId>
+                    <version>1.4.1</version>
                 </dependency>
                 <dependency>
                     <groupId>com.sun.xml.messaging.saaj</groupId>
                     <artifactId>saaj-impl</artifactId>
-                    <version>1.4.0-b03</version>
+                    <version>1.3.23</version>
```
Comment (reta) [inline parent/pom.xml]: The SAAJ (impl) was one of the problems we run into before, so we have now:
 - `1.5.1` in samples/pom.xml
 - `1.3.23` in parent/pom.xml

Meantime, SAAJ API is 1.4.0.1:
 - `<cxf.osgi.saaj.version>[1.4,2)</cxf.osgi.saaj.version>` in parent/pom.xml

Should we keep `1.4.0-b03` to align API and impl? It seemed to work fine.

File: distribution/src/main/release/samples/pom.xml [hunk 2]
```diff
@@ -261,24 +261,24 @@
                     <version>1.3.1</version>
                 </dependency>
                 <dependency>
-                    <groupId>javax.xml.ws</groupId>
-                    <artifactId>jaxws-api</artifactId>
-                    <version>2.3.0</version>
+                    <groupId>jakarta.xml.ws</groupId>
```
Comment (reta) [inline distribution/src/main/release/samples/pom.xml]: This guy brings `jakarta.annotation-api`, should we get rid of `javax.annotation-api` in this case?

File: parent/pom.xml [hunk 3]
```diff
@@ -2854,35 +2878,38 @@
...
-                    <groupId>javax.xml.ws</groupId>
-                    <artifactId>jaxws-api</artifactId>
-                    <version>2.3.1</version>
-                    <exclusions>
-                        <exclusion>
-                            <groupId>javax.xml.bind</groupId>
-                            <artifactId>jaxb-api</artifactId>
-                        </exclusion>
-                    </exclusions>
+                    <groupId>jakarta.xml.ws</groupId>
```
Comment (reta) [inline parent/pom.xml]: This guy brings jakarta.annotation-api, should we get rid of javax.annotation-api in this case?

File: distribution/src/main/release/samples/pom.xml [hunk 4]
```diff
@@ -261,24 +261,24 @@
...
+                    <artifactId>jakarta.xml.ws-api</artifactId>
+                    <version>2.3.2</version>
                 </dependency>
                 <dependency>
```
Comment (reta) [inline distribution/src/main/release/samples/pom.xml]: I think we could remove `javax.activation-api`, `jakarta.activation-api` replaces it.

File: parent/pom.xml (hunk 1 above)
Comment (deki) [reply parent/pom.xml]: Good catch. For me 1.5.1 also works, but 1.4 is fine as well. @ffang what issues did you have?

File: distribution/src/main/release/samples/pom.xml (hunk 4 above)
Comment (deki) [reply distribution/src/main/release/samples/pom.xml]: Definitely, my plan was to do it along with Spring upgrade afterwards. But we can also do it now, it's hard to separate these things.

File: parent/pom.xml (hunk 1 above)
Comment (deki) [reply parent/pom.xml]: Just checked 1.5.2-b01, also works.

Comment (ffang) [reply parent/pom.xml]: @deki , sorry, can't recall exactly, but should be OSGi test related
I suggest we use saaj api 1.4 from SMX and use saaj 1.3 impl

Thanks!

**Pull request merged**